web/styles/left_sidebar.css

```diff
@@ -593,6 +662,88 @@ li.top_left_scheduled_messages {
     grid-template-areas: "starting-offset starting-anchor-element row-content markers-and-controls ending-anchor-element ending-offset";
 }
 
+#views-label-container {
+    grid-template-columns:
+        0 15px minmax(0, 0.5fr) minmax(0, 1fr)
+        20px 20px;
+    /* Base height on 24px icon-target area from Vlad's design. */
+    grid-template-rows: 24px;
+    /* TODO: Rewrite the `--left-sidebar-collapse-widget-gutter`
+       value and the styles dependent on it so that this kind of
+       1990s layout shenanigans is made unnecessary. */
+    margin-left: -3px;
+    cursor: pointer;
+
+    &.showing-expanded-navigation {
+        /* When the expanded navigation is visible,
+           hide the condensed navigation's controls. */
+        #left-sidebar-navigation-list-condensed,
+        .left-sidebar-navigation-menu-icon {
+            display: none;
+        }
+        /* Give the sidebar title through the end of the markers
+           area, if needed. */
+        .sidebar-title {
+            grid-column: row-content-start / markers-and-controls-end;
+        }
+    }
+
+    /* Use a next-sibling combinator (+) to use CSS to show and hide
+       filter rows as needed, based on the narrow. */
+    &.showing-condensed-navigation {
+        + #left-sidebar-navigation-list {
```

Comment:
I suspect that this selector might prove controversial. But 1) even in https://zulip-sidebar.vladkorobov.repl.co/, we expect the header to remain adjacent to the full items, making this less brittle than meets the eye, and 2) I would prefer to avoid the complexity of mucking about in JavaScript with the expanded nav area for things that are happening in the condensed area.

But of course I'm happy to discuss and change.

Reply:
Yeah this seems safe enough; it doesn't seem very likely that someone would be tempted to add something between these elements.